drivers/rtc/rtc_sf32lb.c

```diff
@@ -128,9 +165,164 @@ static int rtc_sf32lb_get_time(const struct device *dev, struct rtc_time *timept
 	return 0;
 }
 
+#ifdef CONFIG_RTC_ALARM
+static int rtc_sf32lb_alarm_get_supported_fields(const struct device *dev, uint16_t id,
+						 uint16_t *mask)
+{
+	ARG_UNUSED(dev);
+
+	if (id != 0 || mask == NULL) {
+		return -EINVAL;
+	}
+
+	*mask = RTC_ALARM_TIME_MASK_HOUR | RTC_ALARM_TIME_MASK_MINUTE | RTC_ALARM_TIME_MASK_SECOND |
+		RTC_ALARM_TIME_MASK_MONTHDAY | RTC_ALARM_TIME_MASK_MONTH |
+		RTC_ALARM_TIME_MASK_WEEKDAY;
+
+	return 0;
+}
+
+static int rtc_sf32lb_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+				     const struct rtc_time *timeptr)
+{
+	const struct rtc_sf32lb_config *config = dev->config;
+	uint32_t alarm_tr;
+	uint32_t alarm_dr;
+	uint32_t alarm_mask = 0;
+
+	if (id != 0) {
+		return -EINVAL;
+	}
+
+	if ((mask > 0) && (timeptr == NULL)) {
+		return -EINVAL;
+	}
+
+	if (!rtc_utils_validate_rtc_time(timeptr, mask)) {
+		return -EINVAL;
+	}
+
+	if (mask == 0) {
```

Comment:
something's wrong with mask here, second check will always be true to start with. is this consistent with supported fields...?

Reply:
```
#define RTC_ALARM_TIME_MASK_SECOND	BIT(0)
```